azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/WebAppBase.java

```diff
@@ -0,0 +1,792 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.website;
+
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+import com.microsoft.azure.management.website.implementation.SiteInner;
+import org.joda.time.DateTime;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An immutable client-side representation of an Azure Web App or deployment slot.
+ */
+public interface WebAppBase<T extends WebAppBase<T>> extends
+        HasName,
+        Refreshable<T>,
+        GroupableResource,
+        Wrapper<SiteInner> {
+
+    /**
+     * @return state of the web app
+     */
+    String state();
+
+    /**
+     * @return hostnames associated with web app
+     */
+    Set<String> hostNames();
+
+    /**
+     * @return name of repository site
+     */
+    String repositorySiteName();
+
+    /**
+     * @return state indicating whether web app has exceeded its quota usage
+     */
+    UsageState usageState();
+
+    /**
+     * @return true if the site is enabled; otherwise, false
+     */
+    boolean enabled();
+
+    /**
+     * @return host names for the web app that are enabled
+     */
+    Set<String> enabledHostNames();
+
+    /**
+     * @return management information availability state for the web app.
+     *
+     * Normal means that the site is running correctly and that
```

Comment:
For auto-generated enums this is the only way right now - Swagger doesn't allow documentations for enums.